resolver/resolver.go

```diff
@@ -372,23 +396,32 @@ func (res *Resolver) handleSRV(rs *records.RecordGenerator, name string, m, r *d
 		}
 
 		m.Answer = append(m.Answer, srvRR)
-		host := strings.Split(srv, ":")[0]
-		if _, found := added[host]; found {
-			// avoid dups
-			continue
+		host, _, err := net.SplitHostPort(srv)
+		if err != nil {
+			logging.Error.Println(err)
 		}
-		if len(rs.As[host]) == 0 {
+		if len(rs.As[host]) == 0 && len(rs.AAAAs[host]) == 0 {
 			continue
 		}
-
-		if a, ok := rs.As.First(host); ok {
-			aRR, err := res.formatA(host, a)
-			if err != nil {
-				errs.Add(err)
-				continue
+		if _, aFound := aAdded[host]; !aFound {
```

Comment:
is it worth using different priorities for ipv4 records vs ipv6 records? via http://ipv6friday.org/blog/2012/01/ipv6-and-dns/

> An IPv4 client should check the lowest priority first and find out that there’s no available host in that priority and switch to the second priority. An IPv6 client finds a host in the lowest priority. A dual stack client will also find the IPv6 host first and stick to that. DNS SRV records are for you to define how you want to be reached.

Reply:
Only the SRV records themselves have the `Priority` field, so this would only make sense if we were creating different SRV hostname records for ipv4 and ipv6 which we aren't currently. The code you commented on is only referring to the `additional section`. So in the example:

```
;; QUESTION SECTION:
;_ipv6-nginx._tcp.marathon.mesos. IN	SRV

;; ANSWER SECTION:
_ipv6-nginx._tcp.marathon.mesos. 60 IN	SRV	0 0 80 ipv6-nginx-6ijia-s1.marathon.mesos.

;; ADDITIONAL SECTION:
ipv6-nginx-6ijia-s1.marathon.mesos. 60 IN A	12.0.2.2
ipv6-nginx-6ijia-s1.marathon.mesos. 60 IN AAAA	fd01:b::2:8000:2
```

Where there are both IPv4 and IPv6 records for a single host (`ipv6-nginx-6ijia-s1.marathon.mesos.`), it would be up to the client to decide which they prefer. Priority would only apply if we were doing something like `ipv6-nginx-6ijia-s1.marathon.mesos.` for A and `ipv6-nginx-6ijia-s1.preferred.marathon.mesos.` for AAAA, and then we would have 2 separate SRV records for `_ipv6-nginx._tcp.marathon.mesos.`.

Reply:
You're right, thanks for clarifying.